src/main/java/ch/heigvd/dai/models/BMPImage.java

```diff
@@ -26,22 +26,57 @@ public void readImage() {
     try (BufferedInputStream bis = new BufferedInputStream(new FileInputStream(inputPath))) {
       header = new BMPHeader();
       header.readHeader(bis);
+      int width = header.getImageWidth();
+      int height = header.getImageHeight();
+
+      image = new Pixel[width][height];
+
+      for (int h = 0; h < height; h++) { // Iterate from top to bottom
+        for (int w = 0; w < width; w++) {
+          int blue = bis.read();
+          int green = bis.read();
+          int red = bis.read();
+          if (blue == -1 || green == -1 || red == -1) {
+            throw new IOException("Unexpected end of pixel data");
+          }
+          // Store pixels from bottom to top
```

Comment:
Why get into all the trouble to write them from bottom to top if afterwards we have to iterate the same way ? Is it necessary ?

Reply:
Because the first bytes in the file are placed at the bottom of the picture, so it is to have a better visual of the picture stored. But it is not necessary.